commands/new_function.go

```diff
@@ -90,20 +90,28 @@ func runNewFunction(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("%s is unavailable or not supported", language)
 	}
 
-	appendMode := len(appendFile) > 0
-	if appendMode {
-		if (strings.HasSuffix(appendFile, ".yml") || strings.HasSuffix(appendFile, ".yaml")) == false {
-			return fmt.Errorf("when appending to a stack the suffix should be .yml or .yaml")
+	appendMode := false
+	if len(yamlFile) > 0 {
```

Comment:
Doesn't this need to be compared with `defaultYAML` to ensure that when a `--yaml` hasn't been passed we aren't writing to `stack.yml` if it exists?